tools/lsp/server_loop.rs

```diff
@@ -35,7 +35,7 @@ const QUERY_PROPERTIES_COMMAND: &str = "queryProperties";
 fn command_list() -> Vec<String> {
     let mut result = vec![];
 
-    #[cfg(any(feature = "preview", target_arch = "wasm32"))]
+    #[cfg(feature = "preview")]
```

Comment:
As we talked before: we need that command for the code lense and code action to work.

Reply:
I just tried the web extension with your patch, and clicking on the "show preview" code lense result in the `command 'showPreview' not found` error

<img width="533" alt="Screenshot_20221102_080127" src="https://user-images.githubusercontent.com/959326/199421915-8fe9302a-ce1a-4f1f-93e5-6d6c71a5aa4a.png">

Reply:
I guess that is because the web extension intercepts the command:

```typescript
   connection.onRequest(async (method, params, token) => {
        if (method === "workspace/executeCommand" && (params as ExecuteCommandParams).command === "showPreview") {
            // forward back to the client so it can send the command to the webview
            return await connection.sendRequest("slint/showPreview", (params as ExecuteCommandParams).arguments);
        }
        return await the_lsp.handle_request(token, method, params);
    });
```

Perhaps the availability of the command could also be advertise by the web extension.

Reply:
For example to injecting the command here:

```typescript
    connection.onInitialize((params: InitializeParams): InitializeResult => {
        the_lsp = slint_lsp.create(params, send_notification, load_file);
        return { capabilities: the_lsp.capabilities() };
    });
```

Reply:
> For example to injecting the command

Yes, that would work...
(on the other hand, why should we filter it out to inject it later? we still expose that command name in the lense)

Reply:
But it does supports it, it's just a "noop".  

> Other editors will choke on that eventually

How so?
What could possibly go wrong if we have this command?

One usecase i'd think of is that not having it would allow the client to detect weather or not the server was build with the option to show the preview builtin or not.  Not sure if that could be usefull to client.

> Instead we should have the plugin itself register the showPreview command: It does create it, after all.

Ok. It just seems a bit complexity for not much gain, but if you really insist, i won't object.

Reply:
> user-visible ugliness

What would be user visible?  The command reported by the LSP server are not user visible, AFAIK.